### PR TITLE
Ticket count email subscribers

### PR DESCRIPTION
**Pull request by shanimal08**
### References Issues:
https://app.asana.com/0/1149047845558135/1132116351826451
### Description:
Ability to add subscribers to the ticket count reports email
### Environment Variables:
 * No change

### CODE REVIEW

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+	}
+
+	refreshSubscribers(eventId) {
+		Bigneon()
+			.events.reports.index({
+				event_id: eventId
+			})
+			.then(response => {
+				this.setState({
+					listData: response.data.data
```
Comment (Jasonvdb) [inline src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: Rather break it out if it's being used more than once `let { data } = response.data`

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+			.catch(error => {
+				console.error(error);
+				notification.showFromErrorResponse({
+					error,
+					defaultMessage: "Could not find list."
+				});
+			});
+	}
+
+	handleNewEmailInput = () => {
```
Comment (Jasonvdb) [inline src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: keep to the standard function way for consistency

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+			previousEmail: false,
+			showFirstInput: false,
+			showNewInput: false,
+			errors: {}
+		};
+
+		this.onClose = this.onClose.bind(this);
+		this.onEnter = this.onEnter.bind(this);
+		this.handleNewEmailInput = this.handleNewEmailInput.bind(this);
+		this.handleRemoveEmail = this.handleRemoveEmail.bind(this);
```
Comment (Jasonvdb) [inline src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: Is there a point to bind this functions if you're still calling them like `this.handleRemoveEmail(listItem.id)`

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+			});
+	};
+
+	onClose() {
+		const { onClose } = this.props;
+		this.setState({ showNewInput: false, showFirstInput: false });
+		onClose();
+	}
+
+	validateEmail = () => {
```
Comment (Jasonvdb) [inline src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: How come you randomly switched to these style of function? 😕

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+			errors,
+			showFirstInput
+		} = this.state;
+		const iconUrlDelete = "/icons/remove-small.png";
+
+		let inputs = [];
+		const newInputs = [];
+
+		if (listData) {
+			if (listData.length > 0) {
```
Comment (Jasonvdb) [inline src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: One if statement will work `if (listData && listData.length > 0) {`

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+									<InputAdornment position="end">
+										<CustomIconButton
+											className={classNames({
+												[classes.btnStyle]: true,
+												[classes.removeBtn]: true
+											})}
+											iconUrl={iconUrlDelete}
+											onClick={() => this.handleRemoveEmail(listItem.id)}
+										>
+											""
```
Comment (Jasonvdb) [inline src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: Does this need to be here?

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+			}
+		}
+		inputs = inputs.concat(newInputs);
+		return inputs;
+	}
+
+	render() {
+		const { eventId, classes } = this.props;
+		const { newSubscriberEmail } = this.state;
+		const iconUrl = "/icons/envelope.png";
```
Comment (Jasonvdb) [inline src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: Who gave you the new icons? Any reason they're now png?

Comment (shanimal08) [reply src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: Karim gave me that one :(

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js (hunk 6 above)
Comment (shanimal08) [reply src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: The Icon button needs 'children'. Should I use a different kind of button?

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js (hunk 7 above)
Comment (Jasonvdb) [reply src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: cool then leave as that for now

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js (hunk 6 above)
Comment (Jasonvdb) [reply src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: If it renders fine without a child change the custom icon to not require it in the props

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js (hunk 4 above)
Comment (shanimal08) [reply src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: There was a reason, that doesn't matter any more 😞 i'm changing them back

File: src/components/pages/admin/reports/counts/TicketCountEmailDialog.js (hunk 5 above)
Comment (shanimal08) [reply src/components/pages/admin/reports/counts/TicketCountEmailDialog.js]: 🤦‍♀

**Pull request merged**